src/org/ods/component/ScanWithSonarStage.groovy

```diff
@@ -1,29 +1,46 @@
 package org.ods.component
 
+import com.cloudbees.groovy.cps.NonCPS
+
+import org.ods.services.BitbucketService
 import org.ods.services.SonarQubeService
 
 class ScanWithSonarStage extends Stage {
   public final String STAGE_NAME = 'SonarQube Analysis'
+  private BitbucketService bitbucket
   private SonarQubeService sonarQube
 
-  ScanWithSonarStage(def script, IContext context, Map config, SonarQubeService sonarQube) {
+  ScanWithSonarStage(def script, IContext context, Map config, BitbucketService bitbucket, SonarQubeService sonarQube) {
     super(script, context, config)
-    if (!config.requireQualityGatePass) {
+    if (config.branch) {
+      config.eligibleBranches = config.branch.split(',')
+    } else {
+      config.eligibleBranches = context.sonarQubeBranch.split(',')
+    }
+    if (!config.containsKey('analyzePullRequests')) {
+      config.analyzePullRequests = true
```

Comment:
true if key does not exist?

Reply:
Yes, this means "enabled by default". I'm checking for the key, otherwise setting it to `false` explicitly would have no effect.